src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,2502 @@
+        # iterate through batch size
+        for pred_probs, pred_mask, target_mask, labels in zip(preds_probs, preds_masks, mask_labels, class_labels):
+            # Compute the classification cost. Contrary to the loss, we don't use the NLL,
+            # but approximate it in 1 - proba[target class].
+            # The 1 is a constant that doesn't change the matching, it can be ommitted.
+            cost_class = -pred_probs[:, labels]
+            # flatten spatial dimension "q h w -> q (h w)"
+            num_queries, height, width = pred_mask.shape
+            pred_mask_flat = pred_mask.view(num_queries, height * width)  # [num_queries, H*W]
+            # same for target_mask "c h w -> c (h w)"
+            num_channels, height, width = target_mask.shape
+            target_mask_flat = target_mask.view(num_channels, height * width)  # [num_total_labels, H*W]
+            # compute the focal loss between each mask pairs -> shape [NUM_QUERIES, CLASSES]
+            cost_mask = pair_wise_sigmoid_focal_loss(pred_mask_flat, target_mask_flat)
+            # Compute the dice loss betwen each mask pairs -> shape [NUM_QUERIES, CLASSES]
+            cost_dice = pair_wise_dice_loss(pred_mask_flat, target_mask_flat)
+            # final cost matrix
+            cost_matrix = self.cost_mask * cost_mask + self.cost_class * cost_class + self.cost_dice * cost_dice
+            # do the assigmented using the hungarian algorithm in scipy
+            assigned_indices: Tuple[np.array] = linear_sum_assignment(cost_matrix.cpu())
+            indices.append(assigned_indices)
+
+        # TODO this is a little weird, they can be stacked in one tensor
+        matched_indices = [
+            (torch.as_tensor(i, dtype=torch.int64), torch.as_tensor(j, dtype=torch.int64)) for i, j in indices
+        ]
+        return matched_indices
+
+    def __repr__(self):
+        head = "Matcher " + self.__class__.__name__
+        body = [
+            f"cost_class: {self.cost_class}",
+            f"cost_mask: {self.cost_mask}",
+            f"cost_dice: {self.cost_dice}",
+        ]
+        _repr_indent = 4
+        lines = [head] + [" " * _repr_indent + line for line in body]
+        return "\n".join(lines)
+
+
+# copied and adapted from original implementation
+class MaskFormerLoss(nn.Module):
+    def __init__(
+        self,
+        num_classes: int,
+        matcher: MaskFormerHungarianMatcher,
+        weight_dict: Dict[str, float],
+        eos_coef: float,
+    ):
+        """
+        The MaskFormer Loss. The loss is computed very similar to DETR. The process happens in two steps: 1) we compute
+        hungarian assignment between ground truth masks and the outputs of the model 2) we supervise each pair of
+        matched ground-truth / prediction (supervise class and mask)
+
+        Args:
+            num_classes (`int`):
+                The number of classes.
+            matcher ([`MaskFormerHungarianMatcher]`):
```

Comment:
```suggestion
            matcher (`MaskFormerHungarianMatcher`):
```
That object won't be in the documentation, so no need to try to resolve the link.